### PR TITLE
add response via api

### PR DESCRIPTION
**Pull request by miltontony**
- [x] add response via api
- [x] increment responses counter
- [x] add validation for numeric field

### CODE REVIEW

File: unicore/ask/service/responses_api.py [hunk 1]
```diff
@@ -15,30 +17,50 @@ def get_response_object(request):
...
+            setattr(response, attr, value)
+        response.question_id = option.question_id
+        option.responses.append(response)
+        self.request.db.flush()
+        return response.to_dict()
 
     @view(renderer='json')
     def collection_get(self):
+        # TODO: find a way to return question responses
+        # and not just option responses
```
Comment (Rizziepit) [inline unicore/ask/service/responses_api.py]: Perhaps allow querystring parameters `option_uuid` and `question_uuid`? I think that filtering logic might come in handy later

File: unicore/ask/service/responses_api.py [hunk 2]
```diff
@@ -15,30 +17,50 @@ def get_response_object(request):
...
+
+    if option is None:
+        raise NotFound
+
+    return option
 
 
 @resource(
-    collection_path='/responses/{question_uuid}',
+    collection_path='/responses/{option_uuid}',
```
Comment (Rizziepit) [inline unicore/ask/service/responses_api.py]: I think this should be `/responses` and then either `question_id` or `option_id` needs to be present in the schema.

Comment (Rizziepit) [inline unicore/ask/service/responses_api.py]: The schema can use a deferred validator along with schema binding to check that one or the other is present. Not sure how that integrates with cornice though.

**Comment by miltontony** [comment]
@Rizziepit ready for review
Took me a while, but I finished it. It's mostly to do with submitting responses via the api

File: unicore/ask/service/responses_api.py [hunk 3]
```diff
@@ -15,29 +19,67 @@ def get_response_object(request):
 
 
 def get_responses(request):
-    question_uuid = request.matchdict['question_uuid']
+    question_uuid = request.validated.get('question_uuid')
+    option_uuid = request.validated.get('option_uuid')
 
-    question = request.db.query(Question).get(question_uuid)
+    if question_uuid:
+        question = request.db.query(Question).get(question_uuid)
```
Comment (Rizziepit) [inline unicore/ask/service/responses_api.py:26]: Is the 404 case being handled?

File: unicore/ask/service/responses_api.py [hunk 4]
```diff
@@ -15,29 +19,67 @@ def get_response_object(request):
...
+        return question.responses
 
-    if question is None:
+    if option_uuid:
+        option = request.db.query(QuestionOption).get(option_uuid)
```
Comment (Rizziepit) [inline unicore/ask/service/responses_api.py:34]: Also 404 case.

**Comment by Rizziepit** [comment]
+1

**Pull request merged**